auth.py

```diff
@@ -307,7 +312,7 @@ def update_profile (user):
                    # If on local environment, we return email verification token directly instead of emailing it, for test purposes.
                    resp = {'username': user ['username'], 'token': hashed_token}
                 else:
-                    send_email_template ('welcome_verify', email, requested_lang (), os.getenv ('BASE_URL') + '/auth/verify?username=' + urllib.parse.quote_plus (username) + '&token=' + urllib.parse.quote_plus (hashed_token))
+                    send_email_template ('welcome_verify', email, requested_lang (), os.getenv ('BASE_URL') + '/auth/verify?username=' + urllib.parse.quote_plus (user['username']) + '&token=' + urllib.parse.quote_plus (hashed_token))
```

Comment:
Wow, great catch. I don't know how this could have ever worked, although I distinctly remember having tested it! Oh well.